### DISCUSSION TITLE
Beta-4 Testing; PHP 7.4 & 8.1; Bug Reports

### DISCUSSION THREAD

**Discussion by vicos59** [geodesicsolutions-community/geocore-community#95] [General]
Notes for my observations will be posted here as I move thru the testing...

Posts with a RED HEART have been resolved.

Posts with EYEBALLS are OPEN.

**Comment by vicos59** [comment]
Might be a good idea to add a link to the Github menu for the releases page, if possible:

https://github.com/geodesicsolutions-community/geocore-community/releases/

I can never seem to find it or remember the link.

**Reply by jonyo** [reply]
It is the "changelog" link since they serve as both the list of downloads and as the changelogs inside github.

**Comment by vicos59** [comment]
Unless I missed it, didn't see instructions on the upgrade instructions page about uploading the optional templates provided on the releases page, e.g. fusion, temp, marquee.

Is it **_easily_** possible to link them to the Github page where they can see the diffs for the template files?

**Reply by jonyo** [reply]
That isn't part of the upgrade...

Those are extra designs, you would start using them when you install the software for the first time, used as the "starting point".

For an update, unless there are specific things that are needed, generally the main page templates don't need to be touched.

**Comment by vicos59** [comment]
PHP 7.4, display_errors on.

Upgrade script ran with no warn/err. I **_believe_** on the final Success page, I clicked on the second link to go to the admin page, but I ended up on the offline page. Going to /admin/ was fine. I didn't back up page to double-check.

https://geodesicsolutions.org/update-instructions/?product=GeoCore+CE+%28Community+Edition%29&version=20.0.0&method=&tplUpdate=export

Step 26 says:

```
Delete the following folders and any contained files from your site:
upgrade/
```

This appears on Manage Addons:

![aaa](https://user-images.githubusercontent.com/6634041/156240577-e09dcc95-82e0-4958-ae89-4d68230e9b29.jpg)


On to testing admin, then client side in 7.4

**Reply by jonyo** [reply]
Fixed in update instructions.

**Comment by vicos59** [comment]
AdPlotter is still showing as an option in Manage Addons. Removed in #72 .

Still allows me to install and enable it.

**Reply by vicos59** [reply]
OK, When I select upgrade from 7.5.1, I see instructions to remove AdPlotter. So, this is a non-issue.

**Comment by vicos59** [comment]
/admin/index.php?page=browsing_filter_settings&mc=addon_cat_core_display

White screen of death. Only this:

` --> Smarty: Unable to load template 'geo_tset:admin/browsing_filter_dependency/select.tpl' in 'addon/core_display/admin/browsing_filter_settings.tpl' <-- `

Addon-->Core Display-->Browsing Filter Settings

**Reply by jonyo** [reply]
Ahh, these are the types of errors we are looking for...  👍

**Reply by vicos59** [reply]
I'm thorough. I document everything. :)

**Reply by jonyo** [reply]
Ok, got this one fixed via #105

**Reply by vicos59** [reply]
Confirmed under 7.4 & 8.1 on Docker

**Comment by vicos59** [comment]
/admin/index.php?page=addon_exporter&mc=addon_cat_exporter

White screen of death. Only this:

` --> Smarty: Unable to load template 'geo_tset:admin/load_settings_table.tpl' in 'addon/exporter/admin/search.tpl' <-- `

Addon-->Listing Exporter-->Export

**Reply by jonyo** [reply]
Looks to be fixed now, LMK if you still see it (after pulling latest changes in `main`).

**Reply by vicos59** [reply]
Confirmed under 7.4, including actual export.

Fails under 8.1. The export page loads, unlike before. However, when you execute the export, using default settings:

```
TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/src/addons/exporter/admin.php:221
Stack trace:
#0 /var/www/html/src/addons/exporter/admin.php(248): addon_exporter_admin->initFeedCriteria(Object(geoListingFeed), Array)
#1 /var/www/html/src/addons/exporter/admin.php(71): addon_exporter_admin->xmlExport(Array, true)
#2 /var/www/html/src/admin/php5_classes/Admin.class.php(469): addon_exporter_admin->update_addon_exporter()
#3 /var/www/html/src/admin/index.php(16): geoAdmin->load_page()
#4 {main}

```

http://localhost:8081/admin/index.php?page=addon_exporter

**Reply by jonyo** [reply]
Ah I didn't try exporting...  I fixed all the errors I saw when exporting using default settings, in #114

**Reply by vicos59** [reply]
I was not thorough enough. While the export created the export.xml file, this is what it contained:

`<pre> --> Smarty: Unable to load template 'geo_tset:admin/export_types/xml.tpl' <-- `

**Reply by jonyo** [reply]
@vicos59 Can you clear your `templates_c/` folder and try again (after pulling latest changes from `main`)?  I tried again and it is working, I even checked the xml contents to be sure.  I see a change that would have fixed the error you got, maybe you didn't have the fix at the time you tested.

**Reply by vicos59** [reply]
Confirmed fixed!

**Comment by vicos59** [comment]
/admin/index.php?page=addon_SEO_main_config&mc=addon_cat_SEO

White screen of death. Only this:

```
SEO Wizard

This SEO First-Time Wizard will guide you through the steps needed to start using search engine friendly URL's on your site. Click continue below to get started.

Template Error: Template file not found!

The template file addon/SEO/HTML/add_button.tpl could not be found in any of the template sets currently loaded.
```

Addon-->SEO Settings

**Reply by jonyo** [reply]
Haha, so my fix for the other spot... Broke this spot...  Or at least, this spot continues to be broken.

I'm re-doing my fix, and now 3rd party addons may need more adjustments in template includes in admin (if they did not already).  I was trying to keep it working as best as possible but its looking like can't have it both ways.

**Reply by jonyo** [reply]
Ok should be fixed.  And re-tested all the other spots (unless I missed one) and they are working.

**Reply by vicos59** [reply]
Page loads under 7.4. Ran through settings, created .htaccess and site is using SEO now.

Appears to work under 8.1 also.

**Comment by vicos59** [comment]
Users/UserGroups-->User Groups Home-->General Group

/admin/index.php?mc=users&page=users_group_edit&c=1

`--> Smarty: Unable to load template 'geo_tset:admin/group_settings.tpl' <-- `

**Reply by jonyo** [reply]
Should be fixed now

**Reply by vicos59** [reply]
7.4: Page loads, saves function.

8.1: Page loads, saves function.

**Comment by vicos59** [comment]
/admin/index.php?mc=users&page=users_view&b=50

` --> Smarty: Unable to load template 'geo_tset:admin/show_tokens_user.tpl' <-- `

Users/Groups-->List Users-->View

**Reply by vicos59** [reply]
7.4: No errors/warnings.

8.1: Page loads with this warning:

`Deprecated: Optional parameter $sql_query declared before required parameter $limit_by is implicitly treated as a required parameter in /var/www/html/src/admin/admin_user_management_class.php on line 1703`

**Reply by vicos59** [reply]
Now works n 8.1.

**Reply by jonyo** [reply]
Edit was broken.  I've been working on it the last few days, finally finished.  See #124 

Note: you may need to change your version in the table `geodesic_version` to 18.02.0 then run the upgrade script.  Otherwise you might be missing a new field and it will cause an error when you try to save a user.

Note to anyone reading this:  this is only needed if you already had version 20 and started from beta 4 or before...  If you do a new install of beta 5 when it comes out it will have it.  Or if you start from 18.02.0 or below and upgrade to beta 5 or higher, you will have it.  This is the down-side to not bothering with beta versioning the version in the code - no way to run SQL "between" beta versions.  But basically, if you aren't sure - you only need to do something if you are on version 20 and can't save when editing a user.

If you want to fix by hand you can also just run this SQL query (if it fails, maybe you already have the column):
```sql
ALTER TABLE `geodesic_userdata`  ADD COLUMN `admin_note` TEXT;
```

**Reply by vicos59** [reply]
Executed SQL statement from Docker CLI for mariadb.

Verified that you can now edit a user and add/change an admin note in both 7.4 & 8.1.

Confirmed fixed!

**Comment by vicos59** [comment]
Not really a bug, more logic. NON-CRITICAL and UNLIKELY...

Users/Groups-->Search Users

If there are 0/zero users, there is still the button to "Send a Message to These Users"

![2](https://user-images.githubusercontent.com/6634041/155012332-330df307-affb-4d40-b0cf-829bc39defd1.JPG)

**Comment by vicos59** [comment]
Payments-->Payment Gateways

/admin/index.php?page=payment_gateways&mc=payments

When you click on 'Configure' for **PayPal Pro**, it says:

`Error: Request failed when attempting to retrieve settings!`

All other gateways expand ok. Of course, the one I use. LOL

**Reply by jonyo** [reply]
This one...  Getting this error on the ajax call:

> <b>Fatal error</b>:  Array and string offset access syntax with curly braces is no longer supported in <b>/var/www/html/src/classes/PEAR/Net/URL.php</b> on line <b>184</b><br />

This is more than a quick fix - it will require #99 updating the PEAR libraries and/or updating paypal pro to use latest libraries.

**Reply by vicos59** [reply]
Probably using Net->URL to call the PayPal URL and fetch the response. Curious why it would need this just to load the admin/config form. That form should only allow input/edit of the user-specific parameters to use the payment gateway.

**Reply by jonyo** [reply]
It's part of the test to be sure paypal pro can be used, I believe.  It is meant to check if it works, and show check marks or x's if it does or does not.  Though now since the call does not work, it doesn't even do that much...

**Reply by jonyo** [reply]
Actually, it is not even calling anything, it's one of the tests that just attempts to include the pear library and if it fails then the test fails.  But it half way works, the file is included, but that causes the error and causes the entire thing to break.

**Reply by jonyo** [reply]
@vicos59 can you confirm you are still seeing this error?  I did a fix that should have at least fixed the admin config, but don't know if the gateway itself works.

And if not I think it would be handled in #56 maybe?

**Reply by vicos59** [reply]
Yes, works in both 7.4 & 8.1! I also did a Save w/o changing anything and it said it saved ok.

Bless you, friend! 

Maybe tomorrow I can try testing actual payment with a PP sandbox account.

**Reply by vicos59** [reply]
Created #138 to perform PP Pro gateway testing.

**Comment by vicos59** [comment]
Some minor layout issue at:

/admin/index.php?page=page_attachments_edit&pageId=1_classified

![3](https://user-images.githubusercontent.com/6634041/155013998-20af7ae1-7cb8-4970-a70e-727eb15e8d43.JPG)

**Brave browser...**

**Reply by jonyo** [reply]
Fixed in #128

**Comment by vicos59** [comment]
/admin/index.php?page=admin_tools_license&licenseRefreshed=1

'Clear Key' and 'Refresh License Data' still there.

Not really a bug, but might confuse some people since it is now obsolete.

![4](https://user-images.githubusercontent.com/6634041/155014996-8fdb17e4-d4ae-4c0c-9039-eb9dd5e1c01e.JPG)

**Reply by jonyo** [reply]
Made #111 for this.

**Comment by vicos59** [comment]
OK, I went thru all of the menu items in admin to make sure they load. Drilled down a few places and did Saves where present. This is all I caught for 7.4. Probably wait for patches before testing /admin/ again with 8.0.

On to the client side.

**Comment by vicos59** [comment]
Changed to 8.0 to test the client side. Reloaded /admin/ and got this. The main client-side page has the same thing. 8.0 testing halted.

```
ValueError: mysqli_stmt::bind_param(): Argument #1 ($types) cannot be empty in /home/[snip]/public_html/vendor/adodb/adodb-php/drivers/adodb-mysqli.inc.php:1241
Stack trace:
#0 /home/[snip]/public_html/vendor/adodb/adodb-php/drivers/adodb-mysqli.inc.php(1241): mysqli_stmt->bind_param('')
#1 /home/[snip]/public_html/vendor/adodb/adodb-php/adodb.inc.php(1539): ADODB_mysqli->_query('SELECT * FROM g...', Array)
#2 /home/[snip]/public_html/vendor/adodb/adodb-php/drivers/adodb-mysqli.inc.php(1135): ADOConnection->_Execute('SELECT * FROM g...', Array)
#3 /home/[snip]/public_html/vendor/adodb/adodb-php/adodb.inc.php(2274): ADODB_mysqli->execute('SELECT * FROM g...', Array)
#4 /home/[snip]/public_html/classes/php5_classes/DataAccess.class.php(458): ADOConnection->GetRow('SELECT * FROM g...', Array)
#5 /home/[snip]/public_html/addons/share_fees/util.php(146): DataAccess->GetRow('SELECT * FROM g...', Array)
#6 /home/[snip]/public_html/addons/share_fees/util.php(36): addon_share_fees_util->getInfo()
#7 /home/[snip]/public_html/classes/php5_classes/Singleton.class.php(53): addon_share_fees_util->__construct()
#8 /home/[snip]/public_html/classes/php5_classes/Addon.class.php(178): Singleton::getInstance('addon_share_fee...')
#9 /home/[snip]/public_html/classes/php5_classes/Addon.class.php(1136): geoAddon->loadCoreEvents()
#10 /home/[snip]/public_html/addons/zipsearch/app_top.php(44): geoAddon::getRegistry('zipsearch')
#11 /home/[snip]/public_html/app_top.common.php(154): require_once('/home/new...')
#12 /home/[snip]/public_html/admin/app_top.admin.php(8): require_once('/home/new...')
#13 /home/[snip]/public_html/admin/index.php(5): require_once('/home/new...')
#14 {main}
```

**Reply by jonyo** [reply]
This one is known, see #89

**Reply by vicos59** [reply]
OK, I was under the impression it ran under 8.0.

**Reply by jonyo** [reply]
it does, just turn off the fee sharing addon until it is fixed.

**Reply by vicos59** [reply]
Not sure if you found this yet... No Issue Disabling **_Share Fees_** Addon. However, when you try to uninstall it:

```
Error: Call to a member function userNotice() on null in /home/[snip]/public_html/addons/share_fees/setup.php:80
Stack trace:
#0 /home/[snip]/public_html/admin/AJAXController/AddonManage.php(100): addon_share_fees_setup->uninstall()
#1 /home/[snip]/public_html/admin/AJAXController/AddonManage.php(33): ADMIN_AJAXController_AddonManage->uninstall('share_fees')
#2 /home/[snip]/public_html/classes/Ajax.class.php(141): ADMIN_AJAXController_AddonManage->_action(Array)
#3 /home/[snip]/public_html/admin/AJAX.php(68): geoAjax->dispatch('AddonManage', '_action', Array)
#4 {main}
```

**Comment by vicos59** [comment]
Client Side under 8.0. Stuff disappears from front page. No visible error msgs.

**PHP 7.4**
![5a](https://user-images.githubusercontent.com/6634041/155028394-91d59516-e231-4a11-af71-52fe56c24a27.JPG)

**PHP 8.0**
![5b](https://user-images.githubusercontent.com/6634041/155028435-04ec8ea9-7e9b-4fe2-b580-f4bf2ed10b5a.JPG)

**Comment by vicos59** [comment]
PHP 8.0

/admin/index.php?page=home

White screen of death:

`Fatal error: Declaration of listing_change_adminOrderItem::geoCart_initItem_new($call_children, $item_type) must be compatible with _listing_placement_commonOrderItem::geoCart_initItem_new($item_type) in /home/aaa/public_html/classes/order_items/listing_change_admin.php on line 192`

**Comment by vicos59** [comment]
Resuming general testing on Docker image.

Using http://localhost:8080/, ran through a user registration , posted ad and edited. Search. Click through the general links.

No issues detected.

**Comment by vicos59** [comment]
Added Issue #113 to update Site Branding.

**Comment by vicos59** [comment]
Here's a new one:

http://localhost:8080/admin/index.php?page=admin_cart&userId=49&action=new&main_type=listing_edit&listing_id=111

` --> Smarty: Unable to load template 'geo_tset:display_cart/index.tpl' in 'admin/cart/index.tpl' <-- 
`

When you click "Edit Details" for a user's listing in admin.

**Reply by jonyo** [reply]
Fixed in #130

**Reply by vicos59** [reply]
Confirmed fixed!

**Comment by vicos59** [comment]
PHP 8.1

http://localhost:8081/category/232/For-Sale.html

```
TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/src/classes/php5_classes/Browse.class.php:869
Stack trace:
#0 /var/www/html/src/classes/php5_classes/Browse.class.php(808): geoBrowse::categoryColumnSort(Array, '4', '0', 9.0)
#1 /var/www/html/src/classes/browse_ads.php(273): geoBrowse->categoryBrowsing()
#2 /var/www/html/src/index.php(782): Browse_ads->browseCategory(0)
#3 {main}

```
This happens from the front page when I click on the 'For Sale' category. There is one ad in that category. Works fine w/7.4

**Reply by jonyo** [reply]
Fixed in #131

**Reply by vicos59** [reply]
Reported bug confirmed fixed!

However I detected another bug on the same page. Will create a new post below for it.

https://github.com/geodesicsolutions-community/geocore-community/discussions/95#discussioncomment-2302066

**Comment by vicos59** [comment]
@jonyo I added 3 new bugs recently.

Also, since there is no way to mark an issue resolved, I marked each post with a HEART if the bug is fixed or a separate issue exists to handle it. Posts marked with a pair of EYEBALLs need attention.

**Reply by jonyo** [reply]
Ahh good system!  So we don't flood issues (since a lot of these end up being grouped anyways - fixed by the same fix).  It will make it easier to go through them.

**Comment by vicos59** [comment]
PHP 8.1

http://localhost:8081/admin/index.php?page=home

In the Software Version iFrame:

```
Error: Non-static method geoSession::currentAdminSeats() cannot be called statically in /var/www/html/src/admin/AJAXController/homeLicense.php:75
Stack trace:
#0 /var/www/html/src/classes/Ajax.class.php(141): ADMIN_AJAXController_homeLicense->getLicenseData(Array)
#1 /var/www/html/src/admin/AJAX.php(68): geoAjax->dispatch('homeLicense', 'getLicenseData', Array)
#2 {main}

```

![aaaaaa](https://user-images.githubusercontent.com/6634041/156802137-654a7ba0-c288-4340-bc46-348813878aa5.JPG)

**Reply by jonyo** [reply]
Fixed by #132

**Reply by vicos59** [reply]
Confirmed fixed!

**Comment by vicos59** [comment]
Also in PHP 8.1

http://localhost:8081/category/237/Antiques/listings/111/Star-Wars-Album.html

Clicked Featured Listing from front page.

```
Deprecated: Optional parameter $db declared before required parameter $category_id is implicitly treated as a required parameter in /var/www/html/src/classes/order_items/_site_class_temp.php on line 162
TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/src/classes/order_items/auction_final_fees.php:862
Stack trace:
#0 /var/www/html/src/classes/order_items/auction_final_fees.php(224): auction_final_feesOrderItem::_initFinalFeeCart()
#1 /var/www/html/src/classes/php5_classes/OrderItem.class.php(1627): auction_final_feesOrderItem::geoCart_initSession_update(NULL)
#2 /var/www/html/src/classes/php5_classes/Cart.class.php(520): geoOrderItem::callUpdate('geoCart_initSes...', NULL)
#3 /var/www/html/src/classes/php5_classes/Cart.class.php(297): geoCart->initSession(0, true)
#4 /var/www/html/src/modules/my_account_links.php(35): geoCart->init(true)
#5 /var/www/html/src/modules/my_account_links.php(283): module_my_account_links_varcache::loadVars()
#6 /var/www/html/src/classes/php5_classes/DataAccess.class.php(1313): require('/var/www/html/s...')
#7 /var/www/html/src/classes/geo_smarty_plugins/function.module.php(16): DataAccess->moduleTag('my_account_link...', Array, Object(Smarty_Internal_Template))
#8 /var/www/html/src/templates_c/309da43765be6e695057dc2111f2cc6ed4b5351e_0.geo_tset.header_tpl.php(118): smarty_function_module(Array, Object(Smarty_Internal_Template))
#9 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_6219b0d52b3064_11349922(Object(Smarty_Internal_Template))
#10 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#11 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#12 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(385): Smarty_Internal_Template->render()
#13 /var/www/html/src/templates_c/a0fd7c11c65e850e5a1bb23bc06f51b94818c436_0.geo_tset.listing_page_tpl.php(57): Smarty_Internal_Template->_subTemplateRender('geo_tset:header...', NULL, NULL, 0, 3600, Array, 0, false)
#14 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_621a8e2305a326_19698604(Object(Smarty_Internal_Template))
#15 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#16 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#17 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(238): Smarty_Internal_Template->render(false, 1)
#18 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(134): Smarty_Internal_TemplateBase->_execute(Object(Smarty_Internal_Template), NULL, NULL, NULL, 1)
#19 /var/www/html/src/classes/php5_classes/View.class.php(431): Smarty_Internal_TemplateBase->display('listing_page.tp...')
#20 /var/www/html/src/classes/site_class.php(426): geoView->render(1)
#21 /var/www/html/src/classes/browse_display_ad.php(1026): geoSite->display_page(false)
#22 /var/www/html/src/index.php(165): Display_ad->display_classified(111)
#23 {main}

```

**Reply by jonyo** [reply]
I seem to have fixed this somewhere already.

**Reply by vicos59** [reply]
Confirmed fixed!

**Comment by vicos59** [comment]
Everything from here forward is after a fresh install of the GeoCoreCE containers in Docker and merging all commits from the main repo to my fork as of around 4pm 05-MAR-2022.

**Comment by vicos59** [comment]
In PHP 8.1 front page won't load.

http://localhost:8081/

```
TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/src/classes/user_management_list_bids.php:214
Stack trace:
#0 /var/www/html/src/modules/my_account_links.php(165): Auction_list_bids->list_auctions_with_your_bid(true)
#1 /var/www/html/src/modules/my_account_links.php(283): module_my_account_links_varcache::loadVars()
#2 /var/www/html/src/classes/php5_classes/DataAccess.class.php(1314): require('/var/www/html/s...')
#3 /var/www/html/src/classes/geo_smarty_plugins/function.module.php(16): DataAccess->moduleTag('my_account_link...', Array, Object(Smarty_Internal_Template))
#4 /var/www/html/src/templates_c/309da43765be6e695057dc2111f2cc6ed4b5351e_0.geo_tset.header_tpl.php(118): smarty_function_module(Array, Object(Smarty_Internal_Template))
#5 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_6219b0d52b3064_11349922(Object(Smarty_Internal_Template))
#6 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#7 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#8 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(385): Smarty_Internal_Template->render()
#9 /var/www/html/src/templates_c/4d9c2a3988e7b1c1f4519600f34d4ae1b8cd0544_0.geo_tset.front_page_tpl.php(57): Smarty_Internal_Template->_subTemplateRender('geo_tset:header...', NULL, NULL, 0, 3600, Array, 0, false)
#10 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_6223caed8b4078_56509030(Object(Smarty_Internal_Template))
#11 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#12 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#13 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(238): Smarty_Internal_Template->render(false, 1)
#14 /var/www/html/src/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(134): Smarty_Internal_TemplateBase->_execute(Object(Smarty_Internal_Template), NULL, NULL, NULL, 1)
#15 /var/www/html/src/classes/php5_classes/View.class.php(431): Smarty_Internal_TemplateBase->display('front_page.tpl')
#16 /var/www/html/src/classes/site_class.php(426): geoView->render(2)
#17 /var/www/html/src/classes/browse_ads.php(874): geoSite->display_page()
#18 /var/www/html/src/index.php(1336): Browse_ads->main()
#19 {main}
```

**Reply by jonyo** [reply]
I believe I fixed this one somewhere too.

**Reply by vicos59** [reply]
Confirmed fixed!

**Comment by vicos59** [comment]
PHP 8.1 (7.4 works fine)

http://localhost:8081/index.php?a=5&b=232&c=0&browse_view=grid

The **grid view** shows nothing when there is 1 ad there. The List and the Gallery views work OK.

![grid](https://user-images.githubusercontent.com/6634041/156906039-6d537199-4232-4321-bbc4-1e3aa4246122.JPG)

**Reply by jonyo** [reply]
The error gets hidden because of where it "stops" in the DOM, but if you view the source you'd see the error.

But not any more, fixed in #136

**Reply by vicos59** [reply]
Confirmed fixed!

**Comment by vicos59** [comment]
@jonyo If you feel most of the bugs are squashed and want to publish Beta-5, I would do a clean install on my test domain and get a few normal users to do some testing to flush out anything we missed. No pressure.

**Comment by vicos59** [comment]
http://localhost:8081/admin/index.php?page=addon_exporter&mc=addon_cat_exporter

![exports777](https://user-images.githubusercontent.com/6634041/156930536-b95e8fc8-f679-4474-8797-e28311dee3b7.JPG)

I checked in `/var/www/html/src/addons/exporter/` and there was no `exports/`

I created one and set it to 777 and the alert goes away.

**Reply by jonyo** [reply]
You probably don't need to set it to 777, just create it.  Probably just add a `.empty` file to it with nothing in it.

Git only tracks files, if there is a totally empty folder it will not be committed, so you have to put a file.  Standard practice is to put the `.empty` like I mentioned, you'll see it in other folders like `user_images` as well.

**Reply by jonyo** [reply]
By the way, what options did you use to see this?  When I export with default settings it seems to work.

**Reply by vicos59** [reply]
It showed up when I loaded the export page:

http://localhost:8081/admin/index.php?page=addon_exporter&mc=addon_cat_exporter

The export worked fine before I created the missing directory.

However, after I created the directory and executed another export, I see the export.xml is there.

If you look under the "Data Exported" tab, you will see that path mentioned.

**Reply by vicos59** [reply]
> Probably just add a .empty file to it with nothing in it.

Do you want me to do this and issue a PR? I verified that the folder is indeed missing from the main repo.

**Reply by jonyo** [reply]
I think maybe I created the folder when I was testing and didn't think much about it so I forgot...

> Do you want me to do this and issue a PR? I verified that the folder is indeed missing from the main repo.

Yeah, if you do a pr you'll also need to add the empty file to `.gitignore` with a `!` in front.  So in `.gitignore` find this line:

```
src/addons/exporter/exports/
```

That tells it to ignore files in that folder.  To make it not ignore the .empty, add this on the next line:


```
!src/addons/exporter/exports/.empty
```

In the .gitignore you'll see it does similar things for the other folders that get stuff added by the software but we don't want to commit it.  Like the templates_c folder.

**Reply by vicos59** [reply]
Created new issue #142

**Comment by vicos59** [comment]
http://localhost:8081/admin/index.php?mc=admin_tools_settings&page=admin_tools_password

Changing admin user/pass. Works in 7.4.

```
Error: Non-static method geoPC::get_hashed_password() cannot be called statically in /var/www/html/src/admin/admin_authentication_class.php:474
Stack trace:
#0 /var/www/html/src/admin/admin_authentication_class.php(519): Admin_auth->update_admin_login(Object(DataAccess), Array)
#1 /var/www/html/src/admin/php5_classes/Admin.class.php(475): Admin_auth->update_admin_tools_password()
#2 /var/www/html/src/admin/index.php(16): geoAdmin->load_page()
#3 {main}

```

**Comment by vicos59** [comment]
Bulk Loader fails at final step of import.

(This is outlined in the PR, but placed here to keep the bugs all in one place: https://github.com/geodesicsolutions-community/geocore-community/pull/148#issuecomment-1062236114)

`example_upload_auctions.txt` fails at the final step:

http://localhost:8081/admin/index.php?page=addon_bulk_uploader_main_config&p=3

```
TypeError: array_merge(): Argument #1 must be of type array, null given in /var/www/html/src/addons/bulk_uploader/admin.php:2007
Stack trace:
#0 /var/www/html/src/addons/bulk_uploader/admin.php(2007): array_merge(NULL, Array)
#1 /var/www/html/src/addons/bulk_uploader/admin.php(1915): addon_bulk_uploader_admin->alterUploadForm()
#2 /var/www/html/src/addons/bulk_uploader/admin.php(416): addon_bulk_uploader_admin->displayBulkUploader()
#3 /var/www/html/src/admin/php5_classes/Admin.class.php(543): addon_bulk_uploader_admin->display_addon_bulk_uploader_main_config()
#4 /var/www/html/src/admin/index.php(16): geoAdmin->load_page()
#5 {main}

```

**Comment by vicos59** [comment]
data_misc.sql execution fails due to mySQL/mariaDB STRICT mode (When SQL was executed directly in DBMS, not during actual install/upgrade script):

/* SQL Error (1364): Field 'attached_user_message' doesn't have a default value */

( This is outlined in the PR, but placed here to keep the bugs all in one place: https://github.com/geodesicsolutions-community/geocore-community/pull/152#issuecomment-1063211200 )
